plugin-server/src/main/ingestion-queues/jobs-consumer.ts

```diff
@@ -0,0 +1,88 @@
+import { StatsD } from 'hot-shots'
+import { EachBatchHandler, Kafka, Producer } from 'kafkajs'
+
+import { KAFKA_EVENTS_DEAD_LETTER_QUEUE, KAFKA_JOBS } from '../../config/kafka-topics'
+import { EnqueuedPluginJob, JobName } from '../../types'
+import { status } from '../../utils/status'
+import { GraphileWorker } from '../graphile-worker/graphile-worker'
+import { instrumentEachBatch, setupEventHandlers } from './kafka-queue'
+
+export const startJobsConsumer = async ({
+    kafka,
+    producer,
+    graphileWorker,
+    statsd,
+}: {
+    kafka: Kafka
+    producer: Producer // NOTE: not using KafkaProducerWrapper here to avoid buffering logic
```

Comment:
I'm not sure about kafkajs and I'm not sure if it's part of the protocol, but the Java Kafka Producer implements linger.ms and libs based off [librdkafka](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md) will all support this as well, as well as some others (at least similar e.g. kafka-python. I would love to use a librdkafka based client, but we should probably try and make do for the time being.

The difference between these and our implementation afaict if that we do not acknowledge when messages are persistent to Kafka, which is quite key I think and was what the previous iteration of this change did.

I haven't run the perf test on without having any buffering. It might be fine for our needs and worth checking out imo.

Reply:
btw one thing that makes me not too worried about using sync produce calls here is that previously we were doing sync graphile / postgresql calls. It's a guess but I would imagine the postgres throughput being worse than Kafka. But let me know if you're not convinced